R/net.core.attr.R

```diff
@@ -0,0 +1,61 @@
+#' @param n.new the number of new nodes to initiate with core attributes
+#' @rdname net-accessor
+#' @export
+append_core_attr <- function(dat, n.new) {
+  dat <- append_attr(dat, "active", 1, n.new)
+  dat <- update_uids(dat, n.new)
+
+  return(dat)
+}
+
+#' @title Create the uids for the new nodes
+#'
+#' @description This function is called by `append_core_attr` and append new
+#' uids to the created nodes. It also keeps track of the already used uids with
+#' the /code{dat[["_last_uid"]]} variable
+#'
+#' @param dat a Master list object of network models
+#' @param n.new the number of new nodes to give \code{uid} to
+#'
+#' @return the Master list object of network models (\code{dat})
+#'
+#' @keywords internal
+update_uids <- function(dat, n.new) {
+  last_uid <- if (is.null(dat[["_last_uid"]])) 0L else dat[["_last_uid"]]
```

Comment:
Is this what initializes the `dat[["_last_uid"]]`? If not, what does? If so, shouldn't this be initialized at the current network size (e.g., `sum(dat$attr$active == 1)`) instead of `0L`?

Reply:
It is, if `dat[["_last_uid"]]` does not exist it sets it to 0L (enforcing integer to reduce the memoy footprint)

On the first call of `append_core_attr` in the initialization module, there are no nodes yet. `n.new` are going to be created. So when `update_uids` is called for the first time, `dat[["_last_uid"]]` must be 0L as no UIDs have been used yet.

The next line creates the first `n.new` UIDs with `seq_len(n.new) + last_uid` (1:n.new in this case as `last_uid == 0L`.

`dat[["_last_uid"]]` is then updated to reflect the fact that UIDs from 1 to n.new are used

Reply:
makes sense now, thanks